connectors/src/connectors/google_drive/temporal/spreadsheets.ts

```diff
@@ -424,7 +425,36 @@ export async function syncSpreadSheet(
     }
   };
 
-  const spreadsheet = await getSpreadsheet();
+  let internalErrorsCount = 0;
+  const maxInternalErrors = 3;
+  let spreadsheet: Awaited<ReturnType<typeof getSpreadsheet>>;
+  // If we consistently get 500 Internal Server Error from Google Sheets after 20 activity failures, we skip the file.
+  for (;;) {
+    try {
+      spreadsheet = await getSpreadsheet();
+      break;
+    } catch (err) {
+      if (err instanceof Error && "code" in err && err.code === 500) {
```

Comment:
Can we move this logic to `getSpreadsheet` so all the logic remains close to the call site?

Reply:
How about I get rid of getSpreadsheet entirely ? That way I can directly return from the `catch`

Reply:
Haha you hate those do you